src/groupByKey.ts

```diff
@@ -2,6 +2,14 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<R>(items: R[], key: keyof R): GroupsMap<R> {
+  const condition: R[keyof R][] = items.map((item: R) => item[key]);
+
+  const groups: GroupsMap<R> = {};
+
+  condition.forEach((el: R[keyof R]) => {
+    groups[el as string] = items.filter((item: R) => item[key] === el);
```

Comment:
also you can resolve this task without using nesting loops,

actually you can resolve whole task using just one loop:

just create a new object, then in forEach, define what group (key) the element should have, and then if you don't already have that group in your new object, create a new array with that key, then push your element there.

If you have any questions - feel free to ask them to the fe_pt2 group!)

Reply:
thanks for your advice)
didn't notice this way of implementation)